Feature/42 integrate wiki

vkurup opened the pull request:
Installed the wiki, integrated it into the site, and set it up to populate the activity stream.

/cc @calebsmith @daaray

vkurup committed:
Add wiki to requirements

vkurup committed:
Add apps to INSTALLED_APPS

vkurup committed:
Add sekizai to TEMPLATE_CONTEXT_PROCESSORS

vkurup committed:
Indentation fixes

vkurup committed:
Added URLs for the wiki

vkurup committed:
Add link to wiki from front page.

vkurup committed:
Begin integration of wiki into main site

- Copy wiki base.html file to local templates
- Modify site base.html file to include css from wiki pages
- Remove wiki header bar and use sitewide one instead
- Include wiki notifications in 'my profile' dropdown

vkurup committed:
Add signal for wiki to populate activity stream

- Since wiki is a third-party app, will include signal in
  raspberryio/signals.py
- Add if-branch in template to get proper url from wiki.URLPath

calebsmith commented on raspberryio/settings/base.py:
```diff
@@ -82,7 +82,7 @@
     'django.contrib.staticfiles.finders.FileSystemFinder',
     'django.contrib.staticfiles.finders.AppDirectoriesFinder',
     'compressor.finders.CompressorFinder',
-#    'django.contrib.staticfiles.finders.DefaultStorageFinder',
```
I think these commented out things in the settings are just leftover from defaults.

I'm not sure if you chose to fix the style here, or if maybe one of your tools did some automatic fixing of the style of these. Of course this is pretty unrelated to the rest of the PR, but I'm up for just removing these commented out settings. If we need the eggs template loader and what not we can always add them back in.

The rest of the settings changes look :100: to me

vkurup replied:
Yeah, I think it's my PEP8 linter. I'll go ahead and remove the commented lines. Thanks!

calebsmith commented on raspberryio/templates/includes/account_action.html:
```diff
@@ -9,6 +9,7 @@
 
 {% if action.target %}
     {% if action.target.get_absolute_url %}<a href="{{ action.target.get_absolute_url }}">{{ action.target }}</a>
+    {% elif action.target.urlpath_set.get.get_absolute_url %}<a href="{{ action.target.urlpath_set.get.get_absolute_url }}">{{ action.target }}</a>
```
I think I understand what you're doing here to make the wiki objects in the activity stream give a proper url but I'm not 100%. I'm wondering, do we know that the urlpath_set will always have only one related object? In other words, is their any risk of a ManyObjectsReturned Exception here? I haven't looked at the model so it could be a non-issue, just want to make sure we aren't introducing a potential failure here.

calebsmith commented on raspberryio/templates/wiki/base.html:
```diff
@@ -0,0 +1,84 @@
+
+  </div> <!-- /container -->
+
+  {% endblock %}
```
I  think addtoblock is something that sekizai has introduced for us? If it works, it would be more consistent to use the extra_js block tag here. I'm not sure what this does but it seems to be the same thing as extra_js used elsewhere.

vkurup committed:
Remove commented lines in base.py settings

vkurup committed:
Use wiki:get url-resolver to get wiki url for activity stream

Getting it via article's urlpath_set is risky because there may be
more than 1 urlpath per article. This shouldn't happen, but there's
nothing in the DB to prevent it.

calebsmith commented:
Great work @vkurup 
:dango: :cake: :candy:
